modules/core_project_factory/main.tf

```diff
@@ -179,7 +179,7 @@ resource "google_service_account_iam_member" "service_account_grant_to_group" {
   compute.networkUser role granted to G Suite group, APIs Service account, and Project Service Account
  *****************************************************************************************************************/
 resource "google_project_iam_member" "controlling_group_vpc_membership" {
-  count = var.enable_shared_vpc_service_project && length(var.shared_vpc_subnets) == 0 ? local.shared_vpc_users_length : 0
+  count = var.grant_services_network_role && var.enable_shared_vpc_service_project && length(var.shared_vpc_subnets) == 0 ? local.shared_vpc_users_length : 0
```

Comment:
Lets add a test to make sure we wont have regressions in the future
https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/test/integration/dynamic_shared_vpc/controls/svpc.rb